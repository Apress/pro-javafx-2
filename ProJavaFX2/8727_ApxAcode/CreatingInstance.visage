/*
 * @test
 * @run
 */

// CreatingInstance.visage
import visage.reflect.*;

public class Point {
  public var x: Number;
  public var y: Number;
  public override function toString() {
    "Point \{ x: {x}, y: {y} \}"
  }
}

public function run() {
  var context = VisageLocal.getContext();
  var classType = context.findClass("CreatingInstance.Point");
  var classValue = classType.allocate();
  classValue.initVar("x", context.mirrorOf(3.0));
  classValue.initVar("y", context.mirrorOf(4.0));
  classValue.initialize();
  var p = (classValue as VisageLocal.ObjectValue).asObject();
  println(p);
}
