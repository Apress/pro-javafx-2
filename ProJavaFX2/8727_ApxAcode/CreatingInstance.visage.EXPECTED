Point { x: 3.0, y: 4.0 }
